Neos.Neos/Classes/Domain/Service/WorkspaceService.php

```diff
@@ -0,0 +1,365 @@
+<?php
+
+/*
+ * This file is part of the Neos.Neos package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\Neos\Domain\Service;
+
+use Doctrine\DBAL\ArrayParameterType;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Exception as DbalException;
+use Neos\ContentRepository\Core\ContentRepository;
+use Neos\ContentRepository\Core\Feature\WorkspaceCreation\Command\CreateRootWorkspace;
+use Neos\ContentRepository\Core\Feature\WorkspaceCreation\Command\CreateWorkspace;
+use Neos\ContentRepository\Core\Projection\Workspace\Workspace;
+use Neos\ContentRepository\Core\SharedModel\ContentRepository\ContentRepositoryId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceDescription as DeprecatedWorkspaceDescription;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceName;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceTitle as DeprecatedWorkspaceTitle;
+use Neos\ContentRepositoryRegistry\ContentRepositoryRegistry;
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Security\Exception\NoSuchRoleException;
+use Neos\Neos\Domain\Model\User;
+use Neos\Neos\Domain\Model\UserId;
+use Neos\Neos\Domain\Model\WorkspaceClassification;
+use Neos\Neos\Domain\Model\WorkspaceDescription;
+use Neos\Neos\Domain\Model\WorkspaceMetadata;
+use Neos\Neos\Domain\Model\WorkspacePermissions;
+use Neos\Neos\Domain\Model\WorkspaceRole;
+use Neos\Neos\Domain\Model\WorkspaceSubjectType;
+use Neos\Neos\Domain\Model\WorkspaceTitle;
+
+/**
+ * @api
+ */
+#[Flow\Scope('singleton')]
+final class WorkspaceService
+{
+    private const TABLE_NAME_WORKSPACE_METADATA = 'neos_neos_workspace_metadata';
+    private const TABLE_NAME_WORKSPACE_ROLE = 'neos_neos_workspace_role';
+
+    public function __construct(
+        private readonly ContentRepositoryRegistry $contentRepositoryRegistry,
+        private readonly UserService $userService,
+        private readonly Connection $dbal,
+    ) {
+    }
+
+    /**
+     * Load metadata for the specified workspace
+     */
+    public function getWorkspaceMetadata(ContentRepositoryId $contentRepositoryId, WorkspaceName $workspaceName): WorkspaceMetadata
```

Comment:
since we are also at `getWorkspacePermissionsForUser` forgiving i would also not throw here at `getWorkspaceMetadata` ... dindt we introduce `WorkspaceClassification::UNKNOWN` for that reason?

Reply:
yes, I agree that this method could always return metadata. `WorkspaceClassification::UNKNOWN` is a bit weird if we use that enum to _create_ or _change_ the classification – but maybe we never need it then (or disallow that enum unit in those cases)